draft-ietf-webpush-protocol.xml

```diff
@@ -527,31 +466,58 @@ Date: Thu, 11 Dec 2014 23:56:55 GMT
-        entity body that is 4k (4096 bytes) or less in size.
-      </t>
 
       <section anchor="request_receipt" title="Requesting Push Message Receipts">
         <t>
-          An application server can use the "urn:ietf:params:push:receipt" link
-          relation type to request a confirmation from the push service when a push
-          message is delivered and acknowledged by the user agent.
+          An application server can include the <xref target="RFC7240">Prefer header
+          field</xref> with the "respond-async" preference to request confirmation
+          from the push service when a push message is delivered and then acknowledged
+          by the user agent.
+       </t>
+        <figure>
+          <artwork type="inline">
+            <![CDATA[
+POST /p/JzLQ3raZJfFBR0aqvOMsLrt54w4rJUsV HTTP/1.1
+Host: push.example.net
+Prefer: respond-async
+TTL: 15
+Content-Type: text/plain;charset=utf8
+Content-Length: 36
+
+iChYuI3jMzt3ir20P8r_jgRR-dSuN182x7iB
+]]>
+          </artwork>
+        </figure>
+        <figure>
+          <preamble>
+            A 202 (Accepted) response indicates that the push message was accepted. A
+            URI for the push message resource that was created in response to the request
+            MUST be returned in the Location header field. The push service MUST also provide
+            a URI for the receipt subscription resource in a link relation of type
+            "urn:ietf:params:push:receipt".
+          </preamble>
+          <artwork type="inline">
+            <![CDATA[
+HTTP/1.1 202 Accepted
+Date: Thu, 11 Dec 2014 23:56:55 GMT
+Link: </r/3ZtI4YVNBnUUZhuoChl6omUvG4ZM>;
+        rel="urn:ietf:params:push:receipt"
+Location: https://push.example.net/d/qDIYHNcfAIPP_5ITvURr-d6BGtYnTRnk
+]]>
+          </artwork>
+        </figure>
+        <t>
+          The application server SHOULD include the returned receipt subscription in future
+          requests for push message delivery with receipts to allow the push service to
+          aggregate receipts. The push service SHOULD return the same receipt subscription in
+          its response, although it MAY return a new receipt subscription if it is unable to
+          reuse the one provided by the application server.
         </t>
         <t>
-          The application sets the link relation value to a receipt subscription URI.
-          This receipt subscription resource MUST be created from the same receipt subscribe
-          resource which was returned with the push message subscription response (see <xref
-          target="message_subscription"/>).
+          A push service MUST return a 400 (Bad Request) status code for requests which contain
+          an invalid receipt subscription. A push service MAY return a 429 (Too Many Requests)
+          status code <xref target="RFC6585"/> to reject receipt requests which omit a receipt
+          subscription.

```

Comment:
I almost didn't get what you were saying with the second one.  I think that we need to be clear on what resource we are safeguarding on the push service by sending 429.  That same point applies to the other paragraph above as well, I guess.

A push service might wish to limit the number of <x> subscriptions that it maintains.  Therefore, a push service MAY return a 429...